libs/remote_storage/src/s3_bucket.rs

```diff
@@ -250,6 +268,77 @@ impl RemoteStorage for S3Bucket {
         Ok(document_keys)
     }
 
+    /// Note: it wont include empty "directories"
+    async fn list_prefixes(
+        &self,
+        prefix: Option<Self::RemoteObjectId>,
+    ) -> anyhow::Result<Vec<Self::RemoteObjectId>> {
+        let list_prefix = match prefix {
```

Comment:
Have few thoughts on this: 
* Join is indeed a bit cleaner, but it will replace only part of the logic. Trailing separator logic will remain in this function
* Regarding `impl RemoteObjectId for PathBuf` here join will clash with the PathBuf::join whci his problematic

Any ideas to make it look better?

Reply:
Maybe it was a too idealistic thought, since indeed the first bullet would require us to distinguish between files and directories, that could take a bit more efforts to implement.
Yet I think it's the `join`'s responsibility to ensure all separators are in place, so not fully agree with that first bullet.

As for the second bullet, it could be just renamed into something else, don't see much issues with that.

But it starts to look like a big complication just for the readability, so there might be a better way to do things here.

Reply:
I mean the last separator. To me there is no sense in join being responsible for path to end with a separator or not. Because this actually changes the API response. So for some cases it may be desired to end with a separator, for some not and IMO join is not the right place to make this decision